tunnel/conn.go

```diff
@@ -196,6 +196,7 @@ func Listen(ctx context.Context, addr string, cert string, targets map[Target]st
 			}()
 			return l, nil
 		}
+		l.Close()
```

Comment:
Can we check the returned error and print to logs if not nil?

Reply:
Hi, thanks for the review.

Sounds good. If we reach here, the returned error should always be non-nil. The error is logged like:

```
2021/06/08 16:36:12 start: failed to create register stream: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:10000: connect: connection refused"
2021/06/08 16:36:12 Tunnel listener will retry in 552.330144ms.
2021/06/08 16:36:13 start: failed to create register stream: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:10000: connect: connection refused"
2021/06/08 16:36:13 Tunnel listener will retry in 1.080381816s.
```

I haven't prefixed the error message with anything as I think the message provides enough detail. We could alternatively prefix it with something like:

```
log.Printf("failed register tunnel listener: %v", err)
```

Reply:
sorry I didn't make myself clear. l.Close() returns an error too. It'd be better to check and print it if non-nil.

Reply:
Apologies, I completely missed the point! Yep, you're right, the error of l.Close() should be checked instead.